docs(refactor): refactor and refresh of filter content (#1366)
kroxylicious/kroxylicious: docs/assemblies/assembly-record-encryption-filter.adoc

// file included in the following:
//
// index.adoc

[id='assembly-record-encryption-filter-{context}']
= Record Encryption filter

[role="_abstract"]

> Streams for Apache Kafka Proxy's Record Encryption filter enhances the security of Kafka messages.

Reviewer: ```suggestion
Kroxylicious's Record Encryption filter enhances the security of Kafka messages.
```

Reviewer 2: Should we be extracting a variable so that `Kroxylicious` can be replaced downstream?

The filter uses industry-standard cryptographic techniques to apply encryption to Kafka messages, ensuring the confidentiality of data stored in the Kafka Cluster.

> Streams for Apache Kafka Proxy centralizes topic-level encryption, ensuring streamlined encryption across Kafka clusters.

Reviewer 2: ```suggestion
Kroxylicious centralizes topic-level encryption, ensuring streamlined encryption across Kafka clusters.
```

> Streams for Apache Kafka Proxy centralizes topic-level encryption, ensuring streamlined encryption across Kafka clusters.

Reviewer 2: `centralizes` or `centralises` ?

Reviewer: Our DEV_GUIDE should call out whether the project is favo(u)ring the orthography of US or standard English.  We should then use it consistently across code and docs.

Reviewer 3: We should use US English for docs and the website, because (I _think_, maybe @PaulRMellor can confirm?) that's the standard used in other Red Hat docs and to which we will be held.

Reviewer 2: _theatrical resigned sigh_

fine.

There are three steps to using the filter:

1. Setting up a Key Management System (KMS).
2. Establishing the encryption keys within the KMS that will be used to encrypt the topics.
3. Configuring the filter within Kroxylicious.

> The filter integrates with a Key Management Service (KMS), which has ultimate responsibility for the safe storage of key material.

Reviewer 2: ```suggestion
The filter integrates with a Key Management Service (KMS), which has ultimate responsibility for the safe storage of sensitive key material.
```

?

or perhaps more extensively
```suggestion
The filter integrates with a Key Management Service (KMS), which has ultimate responsibility for managing the life cycle of sensitive key material.
```

> Currently, the filter integrates with HashiCorp Vault and AWS Key Management Service as its KMS, though further supported KMS integrations are planned.

Reviewer: ```suggestion
Currently, the filter integrates with either HashiCorp Vault or AWS Key Management Service as its KMS. Further KMS integrations are planned.
```

> Currently, the filter integrates with HashiCorp Vault and AWS Key Management Service as its KMS, though further supported KMS integrations are planned.

Reviewer 2: ```suggestion
The filter depends on a KMS implementation, currently HashiCorp Vault and AWS Key Management Service are supported. Users are able to supply implementations for their specific KMS systems. Support for further KMS systems will be based on demand.
```

//overview of the record encryption process
include::../modules/record-encryption/con-record-encryption-overview.adoc[leveloffset=+1]
//setting up hashicorp vault
include::assembly-hashicorp-vault.adoc[leveloffset=+1]
//setting up AWS KMS

> include::assembly-aws-kms.adoc[leveloffset=+1]

Reviewer 2: We really need to be clear here that this is an `X or Y` choice! 

Should we maybe have `Set up a KMS` at `levelOffset=+1` and then the includes at `levelOffset+=2` ?

Reviewer 4: I think it should be shuffled so that we have the generic configuration of the Record Encryption filter at the top level with KMS selections within like:

```
= Setting up the Record Encryption Filter
== Setting up a KMS
=== HashiCorp Vault KMS
=== AWS KMS
```
Though I wonder if we need the `Setting up a KMS` block, or can just say in the `Setting up the Record Encryption Filter` section that you need to pick one only.

= Record Encryption filter

[role="_abstract"]

> Kroxylicious's Record Encryption filter enhances the security of Kafka messages.

Reviewer: Do we want to introduce a variable for Kroxylicious to facilitate the production of derivative works?
